comid/cond_endorse_series_triple_test.go

```diff
@@ -4,44 +4,663 @@
 package comid
 
 import (
+	"encoding/json"
 	"testing"
 
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 	"github.com/veraison/corim/extensions"
 )
 
-func Test_CondEndorseSeries_NewCondEndorseSeriesTriples_OK(t *testing.T) {
+func Test_CondEndorseSeriesTriple_NewCondEndorseSeriesTriples_OK(t *testing.T) {
 	c := NewCondEndorseSeriesTriples()
 	require.NotNil(t, c)
+	assert.True(t, c.IsEmpty())
 }
 
-func Test_CondEndorseSeries_Valid_NOK(t *testing.T) {
+func Test_CondEndorseSeriesTriple_Add_Valid(t *testing.T) {
 	c := NewCondEndorseSeriesTriples()
-	expectedErr := "error at index 0: stateful environment validation failed: environment validation failed: environment must not be empty"
-	series := &CondEndorseSeriesTriple{}
+	// Create a valid triple
+	triple := &CondEndorseSeriesTriple{
+		Condition: ValueTriple{
+			Environment: Environment{
+				Class: &Class{
+					ClassID: MustNewUUIDClassID(TestUUID),
+				},
+			},
+		},
+		Series: *NewCondEndorseSeriesRecords(),
+	}
+
+	c.Add(triple)
+	assert.False(t, c.IsEmpty())
+	err := c.Valid()
+	assert.Contains(t, err.Error(), "no measurement entries")
+}
+
+func Test_CondEndorseSeriesTriple_Valid_EmptyCondition(t *testing.T) {
+	c := NewCondEndorseSeriesTriples()
+	series := &CondEndorseSeriesTriple{
+		Series: *NewCondEndorseSeriesRecords(),
+	}
 	c.Add(series)
 	err := c.Valid()
-	assert.EqualError(t, err, expectedErr)
+	assert.EqualError(t, err, "error at index 0: stateful environment validation failed: environment validation failed: environment must not be empty")
+}
+
+func Test_CondEndorseSeriesTriple_Valid_InvalidCondition(t *testing.T) {
+	c := NewCondEndorseSeriesTriples()
+	series := &CondEndorseSeriesTriple{
+		Condition: ValueTriple{
+			Environment: Environment{
+				Class: &Class{
+					ClassID: MustNewUUIDClassID(TestUUID),
+				},
+			},
+			Measurements: *NewMeasurements().Add(
+				&Measurement{
+					Key: MustNewMkey(uint64(123), UintType),
+				},
+			),
+		},
+		Series: *NewCondEndorseSeriesRecords(),
+	}
+	c.Add(series)
+	err := c.Valid()
+	assert.ErrorContains(t, err, "no measurement value set")
+
+}
+
+func Test_CondEndorseSeriesTriple_Valid_InvalidMeasurements(t *testing.T) {
```

Comment:
What is the difference in the Inputs for this test (line 74) and the one on Line 51?